validator/sawtooth_validator/execution/processor_iterator.py

```diff
@@ -259,3 +259,40 @@ def remove_processor(self, processor_identity):
     def __len__(self):
         with self._lock:
             return len(self._processors)
+
+
+class _CurrentMaxMap(object):
+    """A dictionary that maps a key to a maximum value and a counter value"""
+    def __init__(self):
+        self._dict = dict()
+        self._lock = RLock()
+
+    def get(self, key, default=None):
+        with self._lock:
+            return self._dict.get(key)
+
+    def add_if_unique(self, key, maximum):
+        with self._lock:
+            if not self._dict.get(key):
+                self._dict[key] = {
+                    'maximum': maximum,
+                    'current': 0
+                }
+
+    def inc(self, key):
+        """Increments the current counter for a given key if it is below the
+        maximum count"""
+        with self._lock:
+            if self._dict[key]['maximum'] > self._dict[key]['current']:
```

Comment:
This is a small preference: This statement reads as, generally, "if the maximum is greater than current, do", when the more meaningful way to check would be "If current is less than maximum, do". This is just a small stylistic preference.